optin/puller/workflows/download_postoptin.py

```diff
@@ -31,41 +35,103 @@ def parse_args():
 	return parser.parse_args()
 
 
-async def process_transaction(transaction, nem_client, symbol_client, database, payout_signer_public_keys):
-	transaction_height = transaction['meta']['height']
-
-	nem_network = await nem_client.node_network()
-	symbol_network = await symbol_client.node_network()
-
-	process_result = process_nem_optin_request(nem_network, transaction)
-	if not process_result.is_error:
-		process_result = await check_destination_availability(process_result, nem_client, symbol_client)
-
-	if process_result.is_error:
-		print(f'{transaction_height} ERROR: {process_result.message}')
-		database.add_error(process_result)
-	else:
-		destination_address = symbol_network.public_key_to_address(process_result.destination_public_key)
+class DownloadTransactionProcessor:
+	@classmethod
+	async def create(cls, nem_client, symbol_client, databases):
+		self = DownloadTransactionProcessor()
+		self.nem_client = nem_client
+		self.symbol_client = symbol_client
+		self.databases = databases
+
+		self.nem_network = await nem_client.node_network()
+		self.symbol_network = await symbol_client.node_network()
+		return self
+
+	async def _handle_already_paid(self, nem_transaction, destination_address, payout_transaction_info, process_result):
+		logging.debug(f'handling already paid sym: {destination_address}')
+		nem_transation_datetime = NemNetworkTimestamp(nem_transaction['transaction']['timeStamp']).to_datetime()
+
+		payout_timestamp = SymbolNetworkTimestamp(int(payout_transaction_info.transaction['meta']['timestamp']))
+		payout_datetime = payout_timestamp.to_datetime()
+		if 'testnet' == self.symbol_network.name:
+			payout_datetime = payout_timestamp.to_datetime(SYMBOL_TESTNET_EPOCH_TIME)
+
+		if nem_transation_datetime > payout_datetime:
+			# TODO: what should we do here, mark as duplicate?
+			logging.debug(f'nem transaction time is newer than payout timestamp, payout hash: {payout_transaction_info.transaction_hash}')
+			return False
```

Comment:
@Jaguar0625 I'm not sure if this is ok, this will cause adding only to inprogress as duplicate, so I _guess_ that is fine, if there is next one it will be processed and added to completed